src/HW_models/NRF_EGU.h

```diff
@@ -8,12 +8,70 @@
 #define _NRF_HW_MODEL_EGU_H
 
 #include "nrfx.h"
+#include "hal/nrf_egu.h"
+#include "NRF_PPI.h"
 
 #ifdef __cplusplus
 extern "C"{
 #endif
 
-extern NRF_EGU_Type NRF_EGU_regs[6];
+extern NRF_EGU_Type NRF_EGU_regs[EGU_COUNT];
+extern const ppi_event_types_t PPI_EGU_EVENTS[EGU_COUNT][EGU0_CH_NUM];
+
+/**
+ * @brief return instance number for EGU peripheral
+ *
+ * @param p_reg NRF_EGU_Type pointer
+ * @return Number of EGU instance
+ *
+ */
+#define REG_TO_ID(p_reg)    (((size_t)p_reg - (size_t)(&NRF_EGU_regs[0])) / sizeof(NRF_EGU_Type))
```

Comment:
When placed in this header you are making this macro quite public, but the name is very generic/not specific for the EGU, which is likely to cause confusion and conflicts

Reply:
OK, I will remove those macros from public header

Reply:
Up to you if you want/need to do that or you prefer to rename them with something that is EGU specific